pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -94,6 +130,9 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 					// add to cache if the pod is running or pending
 					pc.update(ip, key)
 				}
+				if !IsPodReady(pod) {
+					ev = model.EventDelete
+				}
```

Comment:
If pod not ready in this branch, return imemdiately

Reply:
i think other handlers also relies on it? This is my understanding from John's comment: 

> We actually do want to fetch the pod before it becomes ready. There are two cases for fetching the pod: (1) we are sending config to that pod, and need info about it.

Reply:
I donot think so. This is the first add, when it is not ready, we could just ignore it because we never have any cache about it.  This is a minor comment though

Reply:
ah, then the update event can push the config to the unready pod.

Reply:
nvm: we could not do this